### PR TITLE
wallet: Give deprecation warning when loading a legacy wallet

### PR DESCRIPTION
**Pull request by achow101**
Next step in legacy wallet deprecation.

### CODE REVIEW

**Comment by dimitaracev** [comment]
ACK [956d05b](https://github.com/bitcoin/bitcoin/pull/27869/commits/956d05bad21478a110a0e798cb2c4b69de62a8bb)

**Comment by glozow** [comment]
Concept ACK

File: src/wallet/wallet.cpp [hunk 1]
```diff
@@ -245,6 +245,11 @@ std::shared_ptr<CWallet> LoadWalletInternal(WalletContext& context, const std::s
             return nullptr;
         }
 
+        // Legacy wallets are being deprecated, warn if the loaded wallet is legacy
+        if (!wallet->IsWalletFlagSet(WALLET_FLAG_DESCRIPTORS)) {
+            warnings.push_back(_("Wallet loaded successfully. The legacy wallet type is being deprecated and support for creating and opening legacy wallets will be removed in the future."));
```
Comment (darosior) [inline src/wallet/wallet.cpp:250]: Maybe hint at how to migrate?

**Review by darosior — COMMENTED** [review]
Concept ACK. How about also logging for wallets that are loaded on startup?

File: src/wallet/wallet.cpp (hunk 1 above)
Comment (achow101) [reply src/wallet/wallet.cpp]: Added a hint.

**Comment by S3RK** [comment]
ACK 2bf7183ea9b42a9edf76bccdf00708341a1d2c0a

**Review by jonatack — COMMENTED** [review]
ACK 2bf7183ea9b42a9edf76bccdf00708341a1d2c0a

Indeed, there is no debug log warning yet as suggested in https://github.com/bitcoin/bitcoin/pull/27869#pullrequestreview-1477293036.

Sanity check:

```json
$ ./src/bitcoin-cli -signet listwallets                 
[
  "sqlite3",
  "bdb"
]
$ ./src/bitcoin-cli -signet unloadwallet bdb            
{
}
$ ./src/bitcoin-cli -signet loadwallet bdb            
{
  "name": "bdb",
  "warnings": [
    "Wallet loaded successfully. The legacy wallet type is being deprecated and support for creating and opening legacy wallets will be removed in the future. Legacy wallets can be migrated to a descriptor wallet with migratewallet."
  ]
}
```

File: test/functional/wallet_backwards_compatibility.py [hunk 2]
```diff
@@ -267,7 +267,8 @@ def run_test(self):
             # Make sure this wallet opens without warnings. See https://github.com/bitcoin/bitcoin/pull/19054
```
[the remark this answers is not in the document]
Comment (achow101) [reply test/functional/wallet_backwards_compatibility.py:267]: updated

**Review by jonatack — COMMENTED** [review]
re-ACK 8fbb6e99bfc85a1b9003cae402a7335843a86abd

modulo https://github.com/bitcoin/bitcoin/pull/27869#pullrequestreview-1477293036

Tested on the command line and with the GUI

Screenshot 2023-07-05 at 10 40 10

**Comment by S3RK** [comment]
reACK 8fbb6e99bfc85a1b9003cae402a7335843a86abd

**Pull request merged**